crates/matrix-sdk-ui/src/room_list/mod.rs

```diff
@@ -97,16 +97,28 @@ impl RoomList {
     ///
     /// A [`matrix_sdk::SlidingSync`] client will be created, with a cached list
     /// already pre-configured.
-    pub async fn new(client: Client) -> Result<Self, Error> {
-        let sliding_sync = client
+    ///
+    /// If `with_encryption` is set to true, then the e2ee and to-device
+    /// extensions will be automatically enabled for this instance of the
+    /// `RoomList` API. It should be set to true if and only if there's no
+    /// `EncryptionSync` API running too.
+    pub async fn new(client: Client, with_encryption: bool) -> Result<Self, Error> {
```

Comment:
I would prefer to get `RoomList::new(Client)` and `RoomList::new_with_encryption(Client)`, so that we don't need to update the tests every time we take a decision on this E2EE situation :-). Thoughts? If it's temporary, then fine, let's keep it as is.